.github/workflows/tests.yml

```
@@ -28,20 +54,70 @@ jobs:
           key: ${{ matrix.job.os }}-m2-${{ hashFiles('**/pom.xml') }}
           restore-keys: ${{ matrix.job.os }}-m2
 
-      - name: Compile and install everything
-        run: mvn clean install -DskipTests
-
       - name: Run Andy unit tests
         run: mvn test -pl andy -DexcludedGroups=assignments
 
+  run_assignment_tests:
+    needs: checkstyle
+    runs-on: ${{ matrix.job.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        job:
+          - os: ubuntu-22.04
+          - os: macos-12
+    if: github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name != github.event.pull_request.base.repo.full_name
+
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v3
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'zulu'
+          java-version: 17
+
+      - name: Cache Maven packages
+        uses: actions/cache@v3
+        with:
+          path: ~/.m2
+          key: ${{ matrix.job.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ matrix.job.os }}-m2
+
       - name: Run Andy in all assignments
         run: mvn test -pl andy -Dgroups=assignments
 
-      - name: Run Weblab runner tests
-        run: mvn test -pl weblab-runner -DexcludedGroups=selenium
+  run_weblab_tests:
```

```suggestion
  run_weblab_runner_tests:
```
I'm changing it because the name is a bit confusing; we also have a different set of tests which interact with WebLab platform itself.